### PR TITLE
Leetcode solution 11 to 16 added

### LINKED ISSUE #325: [Feature Request]: dsa leetcode solution
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

i now it is repretitive task but addiing solution is not as much easy as add problem 
so i am making up to 20  solution in one issue 
 

### Use Case

1. i will give multiple approch
2. in all 2 languages
3. proper description
4. iamges to understand properly

### Benefits

it will add solutions of leetocode

### Add ScreenShots

_No response_

### Priority

Medium

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### LINKED ISSUE #339: [Feature Request]: Adding Sections in Javascript
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

i don't want to make multiple level 2 PRs and issues for 1 or 2 sections 
i woll cover following sections of javascript

1. Js object Constrctuor
2. js array methods
3. js loops
4. js conditional satements and operator
all of these topics are important and od medium or hard category
i will make sure these topics would be easy to understand by my documentation
i will add notes,importatnt highlights and images
to make it simple

### Use Case

1. learning js
2. javascript topics description

### Benefits

i will add multiple topics with quality documentation

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Yashgabani845**
## Related Issue

Closes #325 
## Description
i have added solution fo 6 leetcode problem
each of them contain at least thses sections 

1. problem statement
2. problem description
3. example
4. code in all of 3 languages
5. time and space complexity analysis
6. conclusion

## Type of PR

- [x] Feature enhancement
- [x] Documentation update

## Screenshots / Videos (if applicable)

Screenshot (297)
Screenshot (298)
Screenshot (295)
Screenshot (296)

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

i know these are not as i said in issue but at  that moment 
i did not thought this will take much time
so i have added work which was possible atm
you can label PR whatever this many contribution deserve

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by Yashgabani845** [comment]
@Ajay-Dhangar please review this PR

File: dsa-solutions/lc-solutions/0000-0099/0011-Container-with-most-water.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+- **Time Complexity**: $O(n)$, where n is the length of the array `height`. We traverse the list only once using two pointers from both ends of the array towards the center.
+- **Space Complexity**: $O(1)$, we use a constant amount of extra space for the pointers and variables.
+
+## Conclusion
+
+The two-pointer approach is efficient for solving the Container With Most Water problem, allowing us to explore all possible containers by adjusting the pointers based on the heights of the lines, ensuring that we find the container with the maximum possible area.
+
+
+```
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0011-Container-with-most-water.md:168]: remove  line 166 to 168

````md
```
````

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
only remove one line in 0011-Container-with-most-water.md in botton  text is 

````md
```
````

**Comment by Ajay-Dhangar** [comment]
Most importantly, you should focus on one problem at a time rather than covering multiple issues simultaneously. This way, I can assign you Level 3 tasks for individual DSA problem-solving documentation. By doing so, you can concentrate on the quality of the content rather than the quantity or the number of solutions.

**Comment by Yashgabani845** [comment]
@Ajay-Dhangar i have changed as you said 
and yes  now further i will work only on one issue at a time 
generally i raise new issue after making PR of previous one but then i saw people raising multiple issues 
anyway thanks for giving feedback

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**